### DISCUSSION TITLE
Support for discovering referenced and to-be-closed issues from a PR

### DISCUSSION THREAD

**Discussion by mathstuf** [community/community#24706] [Apps, API and Webhooks]
<p>There doesn’t seem to be an API endpoint (or GraphQL query at least) for what issues are referenced or will be closed when a PR is merged. Am I missing something?</p>

**Comment by lee-dohm** [comment]
<p>I haven’t tested this, but you may be able to go through the <a href="https://developer.github.com/v4/object/pullrequest/#timelineitems" rel="nofollow noopener">timelineItems connection</a> and look for <code>CrossReferencedEvent</code> objects with the <code>willClose</code> field set to <code>true</code>.</p>
<p>Let me know if that helps!</p>

**Comment by klaaspieter** [comment]
<p>That was my initial though as well, but unless I’m mistaking the <code>CrossReferencedEvent</code> <code>willClose</code> flag is called <code>willCloseTarget</code>. In the cross reference the PR is the source and the issue is the target so the flag is actually <code>false</code>.</p>
<p>Perhaps a <code>willCloseSource</code>? :pray:</p>

**Comment by lee-dohm** [comment]
<p>Yes, you’re correct, the field is called <code>willCloseTarget</code>.</p>
<p>Unfortunately, the <code>CrossReferencedEvent</code> object is only added to the timeline of the target, not of the source. So I was mistaken, there isn’t a way to determine the issues that a PR would close via the API. My apologies.</p>

**Comment by dleach02** [comment]
<p>How about after the fact? The Issue in GitHub has a “Fixed by <span class="hashtag">#nnn</span>” status implying that there is an association to a PR from the Issue’s perspective. The timeline will even show the reference and I’m assuming that the merge of the PR invokes an automatic close of the issue referenced in the PR. </p>
<p>I would like to be able to trace back from the issue the PR that closed it similar to how the GitHub UI does it?</p>
<p>I’ve grabbed all of the events to the issue and the Events help page indicates that with the close event:</p>
<pre><code>When the commit_id is present, it identifies the commit that closed the issue using "closes / fixes #NN" syntax.
</code></pre>
<p>But I have found that the commit_id is not populated. Instead, occassionally, there is a reference with a commit ID following the close event. But this doesn’t happen all the time.</p>
<p>So what is being used by the UI to show the status that the issue was closed by a particular PR?</p>

**Comment by mathstuf** [comment, marked as answer]
<p>FWIW, this is now available as the <code>closingIssuesReferences</code> field on <a href="https://docs.github.com/en/graphql/reference/objects#pullrequest" rel="noopener nofollow ugc"><code>PullRequest</code></a> GraphQL objects.</p>

**Comment by mathstuf** [comment]
<p>Alas, the <code>IssuesConnection</code> is optional and not required, but that isn’t that big of a deal; at least the information is available now.</p>
